example/src/main/tut/data/Kleisli.md

```diff
@@ -0,0 +1,144 @@
+    j: Kleisli[F, B, C],
+    k: Kleisli[F, A, B]
+  )(implicit M: Monad[F]): Kleisli[F, A, C] =
+    a => M.flatMap(k(a))(j)
+```
+
+With a little imagination, we can see that the `Kleisli` construction not only forms a new Category, but when M is a `Monad`, `Kleisli`s form a `Monad` as well!
+
+## Functions
+
+We provide the following functions in addition to `wrapKleisli`, `runKleisli`, and `compose` for `Kleisli`:
+
+```
+
+    // turn F into G via some natural transformation
+    def hoist[G[_]](η: F ~> G): Kleisli[G, A, B]
+
+    // analogous to `andThen` for functions
+    def andThen[C](
+      j: Kleisli[F, B, C]
+    )(implicit M: Monad[F]): Kleisli[F, A, C]
+
+    // analogous to `compose` for functions
+    def compose[E](
+      j: Kleisli[F, E, A]
+    )(implicit M: Monad[F]): Kleisli[F, E, B]
+
+    //
+    def first[C](implicit F: Functor[F]): Kleisli[F, (A, C), (B, C)] =
+      Kleisli.first(k)
+
+    def second[C](implicit F: Functor[F]): Kleisli[F, (C, A), (C, B)] =
+      Kleisli.second(k)
+
+    // symbolic version of `andThen`
+    def >=>[C](j: Kleisli[F, B, C])(implicit M: Monad[F]): Kleisli[F, A, C]
+
+    // symbolic version of `compose`
+    def <=<[E](j: Kleisli[F, E, A])(implicit M: Monad[F]): Kleisli[F, E, B]
+
+    // reverse version of `>>=`
+    def =<<(fa: F[A])(implicit M: Monad[F]): F[B]
+
+    // An alias for `andThen`
+    def >>>[C](j: Kleisli[F, B, C])(implicit M: Monad[F]): Kleisli[F, A, C]
+
+    // Split the input between the two argument arrows and combine their output.
+    def ***[C, D](j: Kleisli[F, C, D])(
+      implicit S: Strong[Kleisli[F, ?, ?]],
+      M: Monad[F]
+    ): Kleisli[F, (A, C), (B, D)]
+
+    // Fanout: send the input to both argument arrows and combine their output
+    def &&&[C](j: Kleisli[F, A, C])(
+      implicit S: Strong[Kleisli[F, ?, ?]],
+      M: Monad[F]
+    ): Kleisli[F, A, (B, C)]
+
+```
+
+Usage will vary, but as an example, we will show how some of the symbols may be used:
+
+```tut:silent
+
+import scalaz._
+import Scalaz._ // this reveals Kleisli functions aside from wrapKleisli and runKleisli
+import data._ // this gets us our Kleisli object
+
+val f: Int => List[Boolean] = i => List(i < 2)
+val g: Boolean => List[String] = b => if(b) List("Scalaz") else List("Scala")
+
+val k: Kleisli[List, Int, Boolean] = Kleisli.wrapKleisli(f)
+val j: Kleisli[List, Boolean, String] = Kleisli.wrapKleisli(g)
+
+k >=> j // : Kleisli[List, Int, String]
+
+val l: Kleisli[List, String, Int] = Kleisli.wrapKleisli(s => List(s.size))
+
+k <=< l // : Kleisli[List, String, Boolean]
+
+k =<< List(3) // : List(false): List[Boolean]
+
+```
+## Instances
+
+`Kleisli` arrows are especially interesting, and have an instance for many of the typeclasses in the Scalaz ecosystem. Some of these include:
+
+
+- `Monad` where M is a `Monad`
+
+- `Compose` where M is a `Monad`
+
+- `Monoid` for `Kleisli[M, A, B]` when `M[B]` is any `Monoid`
+
+- `Strong` where `M` is any `Functor`
```

Comment:
`Strong` for `Kleisli[M, ?, ?]`.

These might seem obvious to us, but the target audience are people learning about Kleisli. I hate when I'm learning something new and the prose explaining it doesn't even typecheck.

Reply:
Added